src/mono/wasm/runtime/binding_support.js

```diff
@@ -161,7 +164,18 @@ var BindingSupportLib = {
 			this._interned_js_string_table = new Map ();
 
 			this._js_owned_object_table = new Map ();
-			this._js_owned_object_registry = new FinalizationRegistry(this._js_owned_object_finalized.bind(this));
+			// NOTE: FinalizationRegistry and WeakRef are missing on Safari below 14.1
+			if (typeof globalThis.FinalizationRegistry === "function") {
+				this._js_owned_object_registry = new globalThis.FinalizationRegistry(this._js_owned_object_finalized.bind(this));
+			}
+			if (typeof globalThis.WeakRef !== "function") {
+				// this is trivial WeakRef polyfill, which holds strong refrence, instead of weak one
+				globalThis.WeakRef = function WeakRef(targetObject) {
```

Comment:
I think instead of replacing it in globalThis, we should just locally store our own version of WeakRef to use in the bindings.

Reply:
done